controllers/controllers_suite_test.go

```diff
@@ -63,59 +66,152 @@ func TestControllers(t *testing.T) {
 	RunSpecs(t, "Controller Suite")
 }
 
+type ControllerResetter struct {
+	HetznerClusterReconciler              *HetznerClusterReconciler
+	HCloudMachineReconciler               *HCloudMachineReconciler
+	HCloudMachineTemplateReconciler       *HCloudMachineTemplateReconciler
+	HetznerBareMetalHostReconciler        *HetznerBareMetalHostReconciler
+	HetznerBareMetalMachineReconciler     *HetznerBareMetalMachineReconciler
+	HCloudRemediationReconciler           *HCloudRemediationReconciler
+	HetznerBareMetalRemediationReconciler *HetznerBareMetalRemediationReconciler
+}
+
+func NewControllerResetter(
+	hetznerClusterReconciler *HetznerClusterReconciler,
+	hcloudMachineReconciler *HCloudMachineReconciler,
+	hcloudMachineTemplateReconciler *HCloudMachineTemplateReconciler,
+	hetznerBareMetalHostReconciler *HetznerBareMetalHostReconciler,
+	hetznerBareMetalMachineReconciler *HetznerBareMetalMachineReconciler,
+	hcloudRemediationReconciler *HCloudRemediationReconciler,
+	hetznerBareMetalRemediationReconciler *HetznerBareMetalRemediationReconciler,
+) *ControllerResetter {
+	return &ControllerResetter{
+		HetznerClusterReconciler:              hetznerClusterReconciler,
+		HCloudMachineReconciler:               hcloudMachineReconciler,
+		HCloudMachineTemplateReconciler:       hcloudMachineTemplateReconciler,
+		HetznerBareMetalHostReconciler:        hetznerBareMetalHostReconciler,
+		HetznerBareMetalMachineReconciler:     hetznerBareMetalMachineReconciler,
+		HCloudRemediationReconciler:           hcloudRemediationReconciler,
+		HetznerBareMetalRemediationReconciler: hetznerBareMetalRemediationReconciler,
+	}
+}
+
+var _ helpers.Resetter = &ControllerResetter{}
+
+func (r *ControllerResetter) ResetAndInitNamespace(namespace string, testEnv *helpers.TestEnvironment, t FullGinkgoTInterface) {
+	rescueSSHClient := &sshmock.Client{}
+	rescueSSHClient.Test(t)
```

Comment:
can you maybe comment here what it does?

Reply:
I added this comment:

```go
// ResetAndInitNamespace implements Resetter.ResetAndInitNamespace(). Documentation is on the
// interface.
```

or do you think custom documentation is needed?

Reply:
ok, done.